v1/partial/with_configfile.go

```diff
@@ -0,0 +1,38 @@
+// Copyright 2018 Google LLC All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package partial
+
+import (
+	"github.com/google/go-containerregistry/v1"
+)
+
+// withConfigFile defines the subset of v1.Image used by these helper methods
+// TODO(mattmoor): Consider renaming this to withCore?
+type withConfigFile interface {
+	imageCore
+}
+
+// DiffIDs is a helper for implementing v1.Image
+func DiffIDs(i withConfigFile) ([]v1.Hash, error) {
+	cfg, err := i.ConfigFile()
+	if err != nil {
+		return nil, err
+	}
+	dids := make([]v1.Hash, len(cfg.RootFS.DiffIDs))
+	for i, did := range cfg.RootFS.DiffIDs {
```

Comment:
So the order of this is actually reversed from the cfg.RootFS.DiffIDs field? We might want to make an explicit comment about this somewhere in the v1.Image interface.

Right now it says "matches FSLayers", so I guess that might be clear enough. I'm going to have to look this up everytime anyway...

Reply:
I think the comment belongs on the `DiffIDs` declaration on `v1.Image` specifying the ordering (added).

FWIW, I based the implementation (somewhat blindly) on [this](https://github.com/google/containerregistry/blob/master/client/v2_2/docker_image_.py#L48-L51).